### ISSUE TITLE
Make IsHostLocal check to always return true in RDR3

### ISSUE THREAD

**Issue by Disquse** [citizenfx/fivem#860]
`rage::scriptHandlerNetComponent::IsHostLocal` check always fails in RDR3. This leads to various issues with some natives, such as `GET_CLOSEST_OBJECT_OF_TYPE`, which fail to find non-mission objects.

Need to backport this patch from V:
https://github.com/citizenfx/fivem/blob/d85eee9eb580b4e034d9a03bdf9de292cd1a7759/code/components/gta-net-five/src/NetHook.cpp#L1702

![image](https://user-images.githubusercontent.com/10367215/129933395-2c8585f1-5902-4f96-a224-d740b00f655d.png)

**Comment by MoosheTV** [comment]
Here's a reproduction of this bug:

**Comment by MoosheTV** [comment]
Here's a reproduction of this bug: [https://streamable.com/h0ugx9](https://streamable.com/h0ugx9)

It appears that none of the bool parameters to `GET_CLOSEST_OBJECT_OF_TYPE` makes any impact on whether or not this returns the correct object. In that video, I demonstrate that calling `SET_ENTITY_AS_NO_LONGER_NEEDED` on the object seems to result in `GET_CLOSEST_OBJECT_OF_TYPE` returning nothing, and calling `SET_ENTITY_AS_MISSION_ENTITY` causes it to work again.

I've also tested with both spawned objects and static map objects, as well as networked and non-networked objects, and all of them seem to behave in the same manner.

**Comment by Disquse** [comment]
Linked PR doesn't fix your specific case because it's not a bug. That's how it works. Marked as no longer needed objects can't be interacted with. This works in V in the same way. However the host local check might fix some other issues, so backported anyway.

**Comment by Disquse** [comment]
> 
> 
> As I understood what people said, one flag in V would also return non-script objects.

`isMission` flag works even for marked as no longer needed in V, yeah. I remember I tried to toggle it on and off :\

**Comment by blattersturm** [comment]
`isMission` here has nothing to do with 'being mission objects' but rather to do with being a mission/script host. That's why it ends up with the script host check.

Remember: these arguments are named by community members.

**Comment by Disquse** [comment]
> 
> 
> `isMission` here has nothing to do with 'being mission objects' but rather to do with being a mission/script host. That's why it ends up with the script host check.
> 
> Remember: these arguments are named by community members.

Yeah, I never trust community named arguments, especially random flags :P

However, as said this is not related to `rage::scriptHandlerNetComponent::IsHostLocal` (at least not directly) because if this flag is `false` in RDR3 (in V five works, if passing `false`) it skip this check.

**Comment by Disquse** [comment]
It does more checks later but they all should pass because of the patch. Debugging time I guess.

![image](https://user-images.githubusercontent.com/10367215/130827658-3adc6c4b-ea86-4f24-ad7b-26268b065348.png)

**Comment by blattersturm** [comment]
Irrelevant, argument seems to be unused in RDR compared to V, note V having two conditions that trip host check, where the argument known as `isMission` is not referenced in RDR.
